event-digest/main.go

```diff
@@ -111,45 +142,100 @@ func lineCounter(r io.Reader) (int, error) {
 	return count, nil
 }
 
+func usernameExtractor(r io.Reader, users UsernameSet) error {
+	decoder := json.NewDecoder(r)
+	for {
+		if !decoder.More() {
+			break
+		}
```

Comment:
This seems like an odd way to write a loop. Why did you choose to do it this way rather than a more natural loop conditional? Is this what a `while` loop looks like?